JT/01_environment_setup/README.adoc

```diff
@@ -0,0 +1,409 @@
+
+- config file in the Git directory (that is, _.git/config_) of whatever repository you're currently using: Specific to that single repository. You can force Git to read from and write to this file with the _--local_ option, but that is in fact the default. Unsurprisingly, you need to be located somewhere in a Git repository for this option to work properly.
+
+Each level overrides values in the previous level, so values in .git/config trump those in [path]/etc/gitconfig.
+
+You can view all of your settings and where they are coming from using:
++
+[source,bash]
+----
+$ git config --list --show-origin
+----
+
+**Your Identity**
+
+The first thing you should do when you install Git is to set your user name and email address. This is important because every Git commit uses this information, and it's immutably baked into the commits you start creating:
++
+[source,bash]
+----
+$ git config --global user.name "John Doe"
+$ git config --global user.email john.doe@endava.com
+----
+
+Again, you need to do this only once if you pass the --global option, because then Git will always use that information for anything you do on that system. If you want to override this with a different name or email address for specific projects, you can run the command without the --global option when you're in that project.
+
+**Your default branch name**
+
+By default, Git will create a branch called master when you create a new repository with git init. From Git version 2.28 onwards, you can set a different name for the initial branch.
+
+To set main as the default branch name, do:
++
+[source,bash]
+----
+$ git config --global init.defaultBranch main
+----
+
+If you want to check your configuration settings, you can use the git config --list command to list all the settings Git can find at that point:
++
+[source,bash]
+----
+$ git config --list
+user.name=John Doe
+user.email=john.doe@endava.com
+color.status=auto
+color.branch=auto
+color.interactive=auto
+color.diff=auto
+...
+----
+You can also check what Git thinks a specific key's value is by typing git config <key>:
++
+[source,bash]
+----
+$ git config user.name
+John Doe
+----
+=== [[intellij-idea]]IntelliJ IDEA
+
+. Download and install IntelliJ IDEA
++
+Link for downloads: https://www.jetbrains.com/idea/download/
++
+image::resources/idea-01.png[]
+
+. Set up IntelliJ IDEA
++
+* Open IntelliJ IDEA.
+Upon the first launch, you'll have to accept the JetBrains Community Edition Terms and Privacy Policy. Click on "Accept" to proceed.
++
+image::resources/idea-02.png[]
+
+Then you can see the first screen of IntelliJ IDEA. Here you can:
++
+image::resources/idea-03.png[]
+
+. Install plugins for IntelliJ IDEA
++
+* Gherkin
+* Cucumber for Java
++
+image::resources/idea-04.png[]
++
+* TestNG
```

Comment:
Could not found this plugin in new versions
<img width="674" alt="image" src="https://github.com/bodacd/automation-study-materials/assets/5145950/6f365973-bf93-4fdd-a6de-ac1a2e31fb88">

Reply:
I had checked it and removed that plugin. Thank you.